kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java

```diff
@@ -1,32 +1,37 @@
 package uk.gov.homeoffice.digital.sas.kafka.listener;
 
+import jakarta.persistence.PostPersist;
+import jakarta.persistence.PostRemove;
+import jakarta.persistence.PostUpdate;
 import jakarta.validation.constraints.NotNull;
 import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.message.Messageable;
 import uk.gov.homeoffice.digital.sas.kafka.producer.KafkaProducerService;
 
-public abstract class KafkaEntityListener<T> {
+public class KafkaEntityListener<T extends Messageable> {
 
   private final KafkaProducerService<T> kafkaProducerService;
 
   protected KafkaEntityListener(KafkaProducerService<T> kafkaProducerService) {
     this.kafkaProducerService = kafkaProducerService;
   }
 
-  public abstract String resolveMessageKey(T resource);
-
+  @PostPersist
   protected void sendKafkaMessageOnCreate(T resource) {
```

Comment:
Do we need to re-consider the access modifier here and on the others?
Now that we don't need to directly call any methods as they are listener annotated could it go back to private or even public based on some of our previous discussion?

Reply:
As discussed, I made the methods private, constructor public and class final